control-plane/csi-controller/src/controller.rs

```diff
@@ -0,0 +1,726 @@
+use crate::{ApiClientError, MayastorApiClient};
+use regex::Regex;
+use rpc::csi::*;
+use std::collections::HashMap;
+use tonic::{Response, Status};
+use tracing::instrument;
+use uuid::Uuid;
+
+use common_lib::types::v0::openapi::models::{
+    Node, Pool, PoolStatus, SpecStatus, Volume, VolumeShareProtocol,
+};
+
+use rpc::csi::Topology as CsiTopology;
+
+const K8S_HOSTNAME: &str = "kubernetes.io/hostname";
+const VOLUME_NAME_PATTERN: &str =
+    r"pvc-([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})";
+const PROTO_NVMF: &str = "nvmf";
+const MAYASTOR_NODE_PREFIX: &str = "mayastor://";
+
+#[derive(Debug, Default)]
+pub struct CsiControllerSvc {}
+
+mod volume_opts {
```

Comment:
I don't think we want to (and importantly need to) carry this over from the older implementation. We only support a little number of parameters, right? Some timeout settings, local true/false, etc. 

If we create a struct something like:

```rust
#[derive(Deserialize, Serialize)]
struct VolumeOptions {
   local: bool, 
   io_timeout: u64,
   ... 
}
```

We can then parse the whole parameters thing in one blow, and add nice impls as needed to make the code more readable i.e

```rust
impl VolumeOptions {
pub fn pinned(&self) -> bool {
  true 
}
```

It is more work initially but in the long run it will prevent making us go insane as most of it all is just "does you have this value set?"

Reply:
Good point.